app/src/main/java/org/mifos/mobilebanking/ui/activities/SettingsActivity.java

```diff
@@ -16,12 +18,24 @@ protected void onCreate(Bundle savedInstanceState) {
         setToolbarTitle(getString(R.string.settings));
         showBackButton();
         replaceFragment(SettingsFragment.newInstance(), false, R.id.container);
+        Intent intent = getIntent();
+        if (intent.hasExtra("hasSettingsChanged")) {
+            hasSettingsChanged = intent.getBooleanExtra("hasSettingsChanged",
+                    false);
+        }
     }
 
     @Override
     public void onBackPressed() {
         super.onBackPressed();
-        Intent i = new Intent(this, HomeActivity.class);
-        startActivity(i);
+        if (hasSettingsChanged) {
+            // If Settings have changed, the previous home activity may not be valid
+            // For example if user changes the language we need to
+            // relaunch previous activity again
+
+            ActivityCompat.finishAffinity(this);
```

Comment:
I guess this piece of code is not needed as on pressing the Back button the Activity ends.

Reply:
Yes, but I need to call finishAffinity() for it to work the way I expected. 

Basically, what I have done is that if the user changes the preference for language the hasSettingsChanged boolean becomes true.
If it is true, then after the onbackPressed() call I need to  finish all activities i.e. Settings as well as Home Activity and then launch a new instance of HomeActivity. This new instance will be in the user selected language.